metrics_proto_api_test.go

```diff
@@ -0,0 +1,1138 @@
+			Value: &monitoringpb.TypedValue_DistributionValue{
+				DistributionValue: &distributionpb.Distribution{
+					Count:                 count,
+					Mean:                  mean,
+					SumOfSquaredDeviation: 0,
+					BucketCounts:          bktCounts,
+					BucketOptions: &distributionpb.Distribution_BucketOptions{
+						Options: &distributionpb.Distribution_BucketOptions_ExplicitBuckets{
+							ExplicitBuckets: &distributionpb.Distribution_BucketOptions_Explicit{
+								Bounds: bounds,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createMetricPbPointSummary(count int64, sum float64, percentile []float64, values ...float64) *metricspb.Point {
+	valAtPercentiles := []*metricspb.SummaryValue_Snapshot_ValueAtPercentile{}
+	for i, value := range values {
+		valAtPercentile := &metricspb.SummaryValue_Snapshot_ValueAtPercentile{
+			Value:      value,
+			Percentile: percentile[i],
+		}
+		valAtPercentiles = append(valAtPercentiles, valAtPercentile)
+	}
+
+	return &metricspb.Point{
+		Timestamp: endTimestamp,
+		Value: &metricspb.Point_SummaryValue{
+			SummaryValue: &metricspb.SummaryValue{
+				Count: &wrappers.Int64Value{Value: count},
+				Sum:   &wrappers.DoubleValue{Value: sum},
+				Snapshot: &metricspb.SummaryValue_Snapshot{
+					PercentileValues: valAtPercentiles,
+				},
+			},
+		},
+	}
+}
+
+func createGoogleMetricPbPointInt64(value int64) *monitoringpb.Point {
+	return &monitoringpb.Point{
+		Interval: outInterval,
+		Value: &monitoringpb.TypedValue{
+			Value: &monitoringpb.TypedValue_Int64Value{
+				Int64Value: value,
+			},
+		},
+	}
+}
+
+func createGoogleMetricPbPointDouble(value float64, includeStartTime bool) *monitoringpb.Point {
+	interval := outInterval
+	if includeStartTime == false {
+		interval = outGaugeInterval
+	}
+	return &monitoringpb.Point{
+		Interval: interval,
+		Value: &monitoringpb.TypedValue{
+			Value: &monitoringpb.TypedValue_DoubleValue{
+				DoubleValue: value,
+			},
+		},
+	}
+}
+
+func createMetricPbDescriptor(name, desc, unit string, mt metricspb.MetricDescriptor_Type, lblKeys ...*metricspb.LabelKey) *metricspb.MetricDescriptor {
+	return &metricspb.MetricDescriptor{
+		Name:        name,
+		Description: desc,
+		LabelKeys:   lblKeys,
+		Unit:        unit,
+		Type:        mt,
+	}
+}
+
+func createGoogleMetricPbDescriptor(name, metricType, desc, disp, unit string, mk googlemetricpb.MetricDescriptor_MetricKind, vt googlemetricpb.MetricDescriptor_ValueType, lblKeys ...string) *googlemetricpb.MetricDescriptor {
+	lbls := make([]*labelpb.LabelDescriptor, 0)
+	for _, k := range lblKeys {
+		lbl := &labelpb.LabelDescriptor{
+			Key:       k,
+			ValueType: labelpb.LabelDescriptor_STRING,
+		}
+		lbls = append(lbls, lbl)
+	}
+
+	return &googlemetricpb.MetricDescriptor{
+		Name:        name,
+		Type:        metricType,
+		Labels:      lbls,
+		MetricKind:  mk,
+		ValueType:   vt,
+		Unit:        unit,
+		Description: desc,
+		DisplayName: disp,
+	}
+}
+
+type fakeMetricsServer struct {
```

Comment:
Consider reusing https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/master/equivalence_test.go#L371 and split fake server to a separate file.

Reply:
I cannot reuse that because test is part of stackdriver_test package. However, I plan to consolidate test utility in internal package and then reuse it. But that would be another PR.

Reply:
Make sense to me.